test/icalendar/rrule_test.exs

```diff
@@ -0,0 +1,218 @@
+defmodule ICalendar.RRULETest do
+  use ExUnit.Case
+  alias ICalendar.RRULE
+  doctest ICalendar.RRULE
+
+  describe "deserialize" do
+
+    test "String, Atom & Number properties" do
+      [
+        {"FREQ=DAILY",         %RRULE{frequency: :daily}},
+        {"COUNT=5",            %RRULE{count: 5}},
+        {"INTERVAL=5",         %RRULE{interval: 5}},
+        {"BYSECOND=5",         %RRULE{by_second: [5]}},
+        {"BYSECOND=5,10,15",   %RRULE{by_second: [5, 10, 15]}},
+        {"BYMINUTE=5",         %RRULE{by_minute: [5]}},
+        {"BYHOUR=5",           %RRULE{by_hour: [5]}},
+        {"BYMONTH=5",          %RRULE{by_month: [:may]}},
+        {"BYMONTH=1,3,5",      %RRULE{by_month: [:january, :march, :may]}},
+        {"BYMONTHDAY=5",       %RRULE{by_month_day: [5]}},
+        {"BYMONTHDAY=-5",      %RRULE{by_month_day: [-5]}},
+        {"BYYEARDAY=5",        %RRULE{by_year_day: [5]}},
+        {"BYWEEKNO=5",         %RRULE{by_week_number: [5]}},
+        {"BYSETPOS=5",         %RRULE{by_set_pos: [5]}},
+        {"FREQ=DAILY;COUNT=5", %RRULE{count: 5, frequency: :daily}},
+        {"BYDAY=MO",           %RRULE{by_day: [:monday]}},
+        {"BYDAY=MO,TU,SA",     %RRULE{by_day: [:monday, :tuesday, :saturday]}},
+        {"WKST=MO",            %RRULE{week_start: :monday}},
+        {"x-name=lorem",       %RRULE{x_name: "lorem"}}
+      ]
+      |> Enum.each(fn ({check, result}) ->
+        assert RRULE.deserialize(check) == result
+      end)
+    end
+
+    test "Complex Properties" do
+      rrule =
+        "UNTIL=19970714T133000"
+        |> RRULE.deserialize
+
+      assert rrule.until.time_zone == "Etc/UTC"
+      assert Timex.to_erl(rrule.until) == {{1997, 7, 14}, {13, 30, 0}}
+    end
+
+    test "Validation" do
```

Comment:
It seems that currently we can sort of half parse a RRULE- any unknown or invalid properties are marked as erroneous, and whatever we have left is used to construct the struct.

Is this what we want? Would it be better to fail altogether?